### PR TITLE
[Add Compiler Test] syntax probe

### PR DESCRIPTION
**Pull request by Jim8y**
This pr enables us to convenicently deteact if we support a C# syntax or not without creating a complete unit test or smart contract.

### CODE REVIEW

**Comment by Jim8y** [comment]
Screenshot 2024-08-04 at 17 13 07

File: src/Neo.Compiler.CSharp/CompilationEngine.cs [hunk 1]
```diff
@@ -55,6 +55,35 @@ public CompilationEngine(CompilationOptions options)
...
+                             $"using Neo.SmartContract.Framework.Services;\n" +
+                             $"using System;\n" +
+                             $"using System.Text;\n" +
+                             $"using System.Numerics;\n" +
+                             $"using Neo.SmartContract.Framework;\n\n" +
+                             $"namespace Neo.Compiler.CSharp.TestContracts;\n\n" +
+                             $"public class CodeBlockTest : SmartContract.Framework.SmartContract\n{{\n\n" +
+                             $"    public static void CodeBlock()\n" +
+                             $"    {{\n" +
+                                $"        {codeBlock}\n" +
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/CompilationEngine.cs:70]: I prefer internal, only for developers

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/CompilationEngine.cs:70]: NP, mind to update it for me?

File: tests/Neo.Compiler.CSharp.UnitTests/SyntaxProbes/Helper.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Linq;
+using Akka.Util.Internal;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Neo.Compiler.CSharp.UnitTests.Syntax;
+
+public static class Helper
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/SyntaxProbes/Helper.cs:8]:
```suggestion
internal static class Helper
```

**Review by shargon — APPROVED** [review]

**Pull request merged**